src/cms/i18n.ts

```diff
@@ -0,0 +1,78 @@
+import * as R from 'ramda'
+
+import { Locale } from './i18n-locales'
+
+export const setLocales = <T extends Locale>(locales: T[]) => locales
+
+export interface LocalizedString {
+  locale: string
+  value: string
+}
```

Comment:
(Not a veto)
This is a WithLocale<string>

Your TranslationCollection below is a TranslationCollection<string>

As far as I can tell your functions would just work if you changed the type signatures to be generic

So you could have changed the /src/data/i18n.ts right away instead only changing the system for this single collection. I think we should make these changes globally as soon as possible so that we avoid having multiple parallel systems.

Reply:
agreed, I made it generic.